torch/testing/_internal/common_methods_invocations.py

```diff
@@ -2292,6 +2292,17 @@ def sample_inputs_leaky_relu(op_info, device, dtype, requires_grad):
                requires_grad=requires_grad)) for _ in range(1, N)]
     return tensors
 
+def sample_inputs_avgpool2d(op_info, device, dtype, requires_grad, **kwargs):
+    make_arg = partial(make_tensor, device=device, dtype=dtype, requires_grad=requires_grad)
+
+    cases = (((1, 3, 9, 9), (3, 3), (1, 1), (1, ), True, False, 2),
+             ((1, 1, 4, 4), (2, 2), (1, 1), (0, ), False, True, -2),
+             ((1, 2, 6, 6), (4, 4), (2, 2), (2, ), True, True, 3))
+
+    return [SampleInput(make_arg(input_shape), args=(kernel_size, stride, padding, ceil_mode,
```

Comment:
nit: can you do this with a regular for loop instead of a list comprehension for readability.

Reply:
I really like list comprehensions. I think whether or not to do a list comprehension or do a for-loop should not be something we suggest in reviews because it's a matter of personal taste and there is no official PyTorch Style Guide.

Reply:
Done, thanks for the suggestion, @heitorschueroff!

Reply:
I'm sorry, I saw your comment just now @zou3519! Since I had to add a single case without any args except kernel size as well, so I kept a `generator()` approach for now. Hope that's fine.

Reply:
> I really like list comprehensions. I think whether or not to do a list comprehension or do a for-loop should not be something we suggest in reviews because it's a matter of personal taste and there is no official PyTorch Style Guide.

List comprehensions are great for short one-liners, but I also agree that it is a matter of preferred style and since we don't enforce a style guide it is ok to ignore this suggestion.

Reply:
Either style works, so this is fine :). I wouldn't say that one is nicer than the other because it depends on who is reading the code

Reply:
Thanks for the valuable inputs, @zou3519 and @heitorschueroff! I appreciate it. :)